src/Symfony/Component/HttpKernel/DataCollector/EventDataCollector.php

```diff
@@ -28,94 +28,108 @@
  */
 class EventDataCollector extends DataCollector implements LateDataCollectorInterface
 {
-    private ?EventDispatcherInterface $dispatcher;
+    /** @var iterable<EventDispatcherInterface> */
+    private iterable $dispatchers;
     private ?RequestStack $requestStack;
     private ?Request $currentRequest = null;
 
-    public function __construct(EventDispatcherInterface $dispatcher = null, RequestStack $requestStack = null)
+    /**
+     * @param iterable<EventDispatcherInterface>|EventDispatcherInterface|null $dispatchers
+     */
+    public function __construct(iterable|EventDispatcherInterface|null $dispatchers = [], RequestStack $requestStack = null)
     {
-        $this->dispatcher = $dispatcher;
+        if (!is_iterable($dispatchers)) {
+            $dispatchers = $dispatchers ? ['event_dispatcher' => $dispatchers] : [];
+        }
+        $this->dispatchers = $dispatchers;
         $this->requestStack = $requestStack;
     }
 
     public function collect(Request $request, Response $response, \Throwable $exception = null): void
     {
         $this->currentRequest = $this->requestStack && $this->requestStack->getMainRequest() !== $request ? $request : null;
-        $this->data = [
-            'called_listeners' => [],
-            'not_called_listeners' => [],
-            'orphaned_events' => [],
```

Comment:
to be even more BC, we could keep this structure and add key "dispatchers" to add the new data

Reply:
WDYT of this proposal @MatTheCat? that'd be cleaner to me, BC wise.